Makefile

```diff
@@ -1,8 +1,12 @@
 .PHONY: all
 all: get-deps build
 
+.PHONY: clean-deps
+clean-deps:
+	$(MAKE) -C webui $@
+
 .PHONY: get-deps
-get-deps:
+get-deps: clean-deps
```

Comment:
suggestion: assuming that some of our dependency managers handle preinstalled dependencies, do we want to always clean-deps before getting them?

eg npm doesn't really need us to remove the node_modules dir every time we are adding some dependencies

Reply:
oh just saw Brian's comment :).

Reply:
@hamidzr Up to you, I can see this going either one:
Option 1: we clean-deps before get-deps, we don't call get-deps often and its safer to to clean before running
Option 2: the tools probably do the right thing anyways so getting deps will be faster.

Reply:
I think going with option 1 could also suggest always doing `clean` as part of the `build` targets also. I think option 2 gives us more flexibility but maybe less reliability and wherever we I identify the reliability issue we can prefix it with a clean action? with that said both works for me.

Reply:
@brain-good Would going with option one affect the circle-ci cache?

Reply:
@ybt195 looking ~/.circleci/config.yml, for each webui get-deps, we have a check to see if `node_modules` directory exist in the cache and if it does we skip the `get-deps` target. So from the circleCI POV we might be ok with either choice. The other thing is, we are using `npm ci` to download our dependencies defined by our `package-lock.json` file and it will also wipe out the `node_modules` directory for each `make get-deps/npm ci` call anyways. My preference is however option 2 to avoid having another layer of wiping out `node_modules` and trying to keep `get-deps` to just handle on getting dependencies.

Reply:
suggestion: drop `clean-deps` from `get-deps` unless something besides `webui` needs it.
I should mention all 3 webui targets (react, elm, tests) are using `npm ci` which drops its `node_modules` directory before each dependencies fetch, so `webui` definitely doesn't need it.